restli-int-test/src/test/java/com/linkedin/restli/examples/TestParseqBasedFluentClientApi.java

```diff
@@ -644,6 +659,200 @@ public void testAssociateResourceBatchPartialUpdate() throws Exception
+      try{
+        Assert.assertEquals(
+        subgreetingsFluentClient.get(1L).toCompletableFuture().get(5000, TimeUnit.MILLISECONDS),
+            greeting
+        );
+      }
+      catch (Exception e)
+      {
+        Assert.fail("Unexpected error");
+      }
+    });
+    updateStage.toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
+    Assert.assertFalse(updateStage.toCompletableFuture().isCompletedExceptionally());
+
+    // Partial update
+    Greeting update = greeting.copy();
+    String partialUpdateMessage = "Partial update message";
+    update.setMessage(partialUpdateMessage);
+    CompletionStage<Void> partialUpdateResult = subgreetingsFluentClient.partialUpdate(1L, PatchGenerator.diff(greeting, update));
+    partialUpdateResult.toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
+    Assert.assertEquals(subgreetingsFluentClient.get(1L).toCompletableFuture().get(500, TimeUnit.MILLISECONDS).getMessage(), partialUpdateMessage);
+    Assert.assertFalse(partialUpdateResult.toCompletableFuture().isCompletedExceptionally());
+
+
+    // create
+    String msg = Double.toString(Math.random());
+    CompletionStage<Long> result = subgreetingsFluentClient.create(getGreeting(msg));
+    CompletableFuture<Long> createFuture = result.toCompletableFuture();
+    Long createdId = createFuture.get(5000, TimeUnit.MILLISECONDS);
+    Assert.assertTrue(subgreetingsFluentClient.get(createdId).toCompletableFuture().get(5000, TimeUnit.MILLISECONDS).getMessage().equals(msg));
+
+    // batch create
+    String msg1 = Double.toString(Math.random());
+    String msg2 = Double.toString(Math.random());
+    CompletionStage<List<CreateIdStatus<Long>>>
+        batchCreateStage = subgreetingsFluentClient.batchCreate(Arrays.asList(getGreeting(msg1), getGreeting(msg2)));
+    CompletableFuture<List<CreateIdStatus<Long>>> batchCreateFuture = batchCreateStage.toCompletableFuture();
+    List<CreateIdStatus<Long>> createdList = batchCreateFuture.get(5000, TimeUnit.MILLISECONDS);
+    CompletionStage<Map<Long, EntityResponse<Greeting>>> batchGetStage =
+        subgreetingsFluentClient.batchGet(createdList.stream().map(CreateIdStatus::getKey).collect(Collectors.toSet()));
+
+    Map<Long, EntityResponse<Greeting>> entities = batchGetStage.toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
+    Assert.assertEquals(entities.size(), 2);
+  }
+
+  /**
+   * Test {@link com.linkedin.restli.examples.greetings.server.SimpleResourceUnderCollectionResource}
+   * A complete set of request tests were tested in {@link TestSimpleResourceHierarchy}
+   */
+  @Test public void testSubResource_oneLayerPathKey() throws Exception
+  {
+    // Get
+    SubsubgreetingFluentClient subsubClient = new SubsubgreetingFluentClient(_parSeqRestliClient, _parSeqUnitTestHelper.getEngine());
+    Greeting greeting = subsubClient.withSubgreetingsId(1L).get().toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
```

Comment:
Can you also add a test for getting the sub resource client using:

parentClient.subResourceOf(parentId)

Reply:
@karthikbalasub I didn't plan to add it here.

I had a separate task in the backlog for "wiring the interface", and I will add test there.